test/cluster.sh

```diff
@@ -62,11 +62,20 @@ function install_knative_serving() {
   echo "Istio CRD YAML: ${INSTALL_ISTIO_CRD_YAML}"
   echo "Istio YAML: ${INSTALL_ISTIO_YAML}"
   echo "Knative YAML: ${INSTALL_RELEASE_YAML}"
+
   echo ">> Bringing up Istio"
   kubectl apply -f "${INSTALL_ISTIO_CRD_YAML}" || return 1
   kubectl apply -f "${INSTALL_ISTIO_YAML}" || return 1
 
   echo ">> Bringing up Serving"
+  # Delete existing knative-ingressgateway deployments and services if they are not included in this version.
+  # This is so that the upgrade tests can pass between versions that have knative-ingressgateway and versions
+  # that don't.
+  # TODO(nghia): Remove this when knative-ingressgateway is removed.
+  if ! kubectl create -f "${INSTALL_RELEASE_YAML}" --dry-run -lknative=ingressgateway; then
```

Comment:
not sure about -lknative, but as long as you are sure.

Reply:
Yes, that's the label selector we use for the `knative-ingressgateway` service.  I tested that locally

```
nghia@nghia:~/go/src/github.com/knative/serving [add-gateway-cleanup]
$ kubectl create -f config/ --dry-run -lknative=ingressgateway
service/knative-ingressgateway created (dry run)
deployment.extensions/knative-ingressgateway created (dry run)
nghia@nghia:~/go/src/github.com/knative/serving [add-gateway-cleanup]
$ echo $?
0
```